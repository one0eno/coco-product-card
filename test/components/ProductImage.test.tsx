// import * as React from 'react';
// import * as ReactDOM from 'react-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('Debe de mostrar el url de la imagen ', () => {
    //config
    const wrapper = renderer.create(
      <ProductImage img="https://hola.jpb" className="custom-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
    //result
  });

  test('Debe mostrar el componente con la imagen del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
