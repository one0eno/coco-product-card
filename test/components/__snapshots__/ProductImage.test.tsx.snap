// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el url de la imagen  1`] = `
<img
  alt="Product image"
  className="productImg custom-class"
  src="https://hola.jpb"
/>
`;

exports[`ProductImage Debe mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg "
    src="coffee-mug.png"
  />
</div>
`;
